spec/features/admin/admins/admin_create_admin_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+feature 'Admin can create a new User' do
+
+  signed_in_as(:admin) do
+    before do
+      click_header_option("Dashboard")
+      click_sidemenu_option("Admins")
+      click_link("Add new admin")
+    end
+
+    scenario 'Admin creates user with valid data' do
+      fill_in("Email", with: "valid@example.com")
+      fill_in("Password", with: "password")
+      click_button("Create")
+
+      # Current user should be redirected to the index
+      expect(current_path).to eq(admin_admins_path)
+
+      # User should be saved
+      latest_user = User.order(:created_at).last
+      expect(latest_user.email).to eq("valid@example.com")
+      expect(latest_user).to be_admin
+    end
+
+    scenario 'Admin creates user with invalid data' do
+      fill_in("Email", with: "")
+      fill_in("Password", with: "")
+      click_button("Create")
+
+      # Ensure no user is created
+      expect(User.count).to eq(1)
+      expect(page).to have_error_message("user_email", "can't be blank")
```

Comment:
[PP]

Feature tests are sometimes temperamental, if you run this test 100 times, it might fail 10 times, not because you implementation is incorrect, but because your test might be running faster than your server responses.

So this is the current workflow:
```ruby
click_button("Create") #A
expect(User.count).to eq(1) #B
expect(page).to have_error_message("user_email", "can't be blank") #C
```

Believe it or not by `#B` can run before `#A` process has finished. If that happens, your test will blow up because `User.count` will return 0.

Every time you expect something in the page to appear, like an error message, a flash message, a certain content, etc... capybara automatically waits till the content shows up to finally continue your test assertions.

So in order to prevent false negatives, you could do:

```ruby
click_button("Create") #A
expect(page).to have_error_message("user_email", "can't be blank") #B
expect(User.count).to eq(1) #C
```

As soon as you hit `#A`, Capybara will wait for the content `can't be blank` to show up, so `#B` will automatically hang for while until the success page is rendered.

After `#B` has finished you can assert `expect(User.count).to eq(1)` more precisely, preventing false negatives.